src/crypto/public_key/rsa.js

```diff
@@ -261,15 +225,23 @@ export async function generate(bits, e) {
   // RSA keygen fallback using 40 iterations of the Miller-Rabin test
   // See https://stackoverflow.com/a/6330138 for justification
   // Also see section C.3 here: https://nvlpubs.nist.gov/nistpubs/FIPS/NIST
-  let q = await randomProbablePrime(bits - (bits >> 1), e, 40);
-  let p = await randomProbablePrime(bits >> 1, e, 40);
+  let p;
+  let q;
+  let n;
+  do {
+    q = await randomProbablePrime(bits - (bits >> 1), e, 40);
+    p = await randomProbablePrime(bits >> 1, e, 40);
+    n = p.mul(q);
+  } while (n.bitLength() !== bits);
+
+  const phi = p.dec().imul(q.dec());
 
   if (q.lt(p)) {
     [p, q] = [q, p];
   }
-  const phi = p.dec().imul(q.dec());
+
```

Comment:
I get that it doesn't matter, but I think it was clearer the other way around - then it's sort of more obvious that flipping p and q doesn't cause any issues.

Reply:
I had the opposite impression, namely that doing it afterwards implied the order mattered 😄  and in the WebCrypto code above we also swap the numbers when returning, so it's consistent with that.

Reply:
OK, fair enough :+1: